### PR TITLE
[etrog] Info Root SMT

### PR DESCRIPTION
**Pull request by revitteth**

### CODE REVIEW

File: smt/pkg/blockinfo/block_info.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
+package blockinfo
+
+import (
+	"github.com/ledgerwatch/erigon/smt/pkg/smt"
+	"math/big"
+)
+
+//func FillReceiptTree(
```
Comment (mandrigin) [inline smt/pkg/blockinfo/block_info.go]: hmm, why do we need this commented-out code?

File: smt/pkg/blockinfo/block_info_test.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
+package blockinfo
+
+import (
+	"github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon/smt/pkg/smt"
+	"math/big"
+	"testing"
+)
+
+//func TestBlockInfo(t *testing.T) {
```
Comment (mandrigin) [inline smt/pkg/blockinfo/block_info_test.go]: why do we need this commented-out code?

File: smt/pkg/blockinfo/keys.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+const IndexBlockHeaderParamBlockHash = 0
+const IndexBlockHeaderParamCoinbase = 1
+const IndexBlockHeaderParamNumber = 2
+const IndexBlockHeaderParamGasLimit = 3
+const IndexBlockHeaderParamTimestamp = 4
+const IndexBlockHeaderParamGer = 5
+const IndexBlockHeaderParamBlockHashL1 = 6
+const IndexBlockHeaderParamGasUsed = 7
+
+const IndexBlockHeaderParam = 7
```
Comment (mandrigin) [inline smt/pkg/blockinfo/keys.go]: why is this also 7 and what does this one mean?

**Review by mandrigin — CHANGES_REQUESTED** [review]
a few minor changes requested

File: smt/pkg/blockinfo/block_info_test.go (hunk 2 above)
Comment (revitteth) [reply smt/pkg/blockinfo/block_info_test.go]: WIP - implemented

File: smt/pkg/blockinfo/block_info.go (hunk 1 above)
Comment (revitteth) [reply smt/pkg/blockinfo/block_info.go]: WIP - implemented.

File: smt/pkg/blockinfo/keys.go (hunk 3 above)
Comment (revitteth) [reply smt/pkg/blockinfo/keys.go]: JS implementation copy pasta, fixed now

**Review by mandrigin — APPROVED** [review]

**Pull request merged**